astlib/location.mli

```diff
@@ -40,3 +40,19 @@ val update
   -> ?ghost : bool
   -> t
   -> t
+
+type location = t
+
+module Error : sig
+  type t
+
+  val make :
+    loc: location ->
+    fmt_msg: (Format.formatter -> unit) ->
+    unit ->
```

Comment:
I suggest this API which is slightly simpler:

```ocaml
val make : loc:location -> (Format.formatter -> unit) -> unit
```

Reply:
I wouldn't call that `make` if it returns a unit, but perhaps `report`. At which point we don't need a submodule at all, but just `Location.report_error` or some such. I agree that we don't need two steps and an indirection via `Error.t`.

Reply:
I think the `unit` in @diml suggestion was a typo and that he meant `t`.

Just to clarify you're suggesting we get rid of the abstract type and add arguments to the `report` and `to_extension` right?

I agree that right now the indirection isn't necessary but I feel like it's more future proof that way. If we need to add new functions there, we're going to have to duplicate the arguments for each of them and ultimately it's just going to end up like `Location`, a bunch of slightly different functions we try to maintain stable. I feel like an abstract type is a better way to deal with this.
 
Also @diml suggested that at some point we'll need a way to convert from extension points to errors. I'm guessing that's only necessary when we wish to report them so we could add a report function that takes an `extension` as input but at this point it would roughly correspond to using `extension` as an unsafe and exposed type for errors.

Again since it's all going to be wrapped by `ppx` it's not too much of an issue but I'm still not sure that's the best way to go.

Reply:
That was indeed a typo. The number 1 priority for `Astlib` is that it is easy to preserve the interface when the compiler changes. If the API is a bit clunky to use, that's fine, we'll just add our own nicer API on top.